RadarSDK/RadarAPIClient.m

```diff
@@ -155,6 +156,16 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
     params[@"uaNamedUserId"] = [RadarUtils uaNamedUserId];
     params[@"uaSessionId"] = [RadarUtils uaSessionId];
     params[@"source"] = [Radar stringForSource:source];
+    RadarTripOptions *tripOptions = [RadarSettings tripOptions];
+    if (tripOptions) {
+        params[@"tripExternalId"] = tripOptions.externalId;
+        params[@"tripMetadata"] = tripOptions.metadata;
+        if (tripOptions.destinationGeofenceTag && tripOptions.destinationGeofenceExternalId) {
+            params[@"tripDestinationGeofenceTag"] = tripOptions.destinationGeofenceTag;
+            params[@"tripDestinationGeofenceExternalId"] = tripOptions.destinationGeofenceExternalId;
+            params[@"tripMode"] = [Radar stringForMode:tripOptions.mode];
+        }
+    }
```

Comment:
wonder if we shall put them under 

```
params[@"tripOptions"] = @{
@"externalId": xxx ///
@"mode": yyy ///
...
}
```

Reply:
Open to this, let's discuss tradeoffs when we build the server piece

Reply:
https://github.com/radarlabs/server/blob/master/app.ts#L1002
https://github.com/radarlabs/server/blob/master/server/services/UserUpsertService.ts#L335
Server expects a `tripOptions` dictionary wrapping all trip related fields. The main benefit of this is not to bloat the userUpsertService interface, and make it easier for validity checks

```
params[@"tripOptions"] = @{
@"externalId": ... // required
@"mode": ... // required
@"destinationGeofenceTag": ...
@"destinationGeofenceExternalId": ...
@"metadata": ...
}
```

Reply:
Awesome, I will update this